File: tests/c-tests/write_stderr.c
// SPDX-License-Identifier: Apache-2.0

#include "libc.h"

int main(void) {
    const char msg[] = "hi\n";
    const int len = sizeof(msg) - 1;
    return write(STDERR_FILENO, msg, len) != len;
}

File: tests/rust_syscall_tests/src/bin/write_stderr.rs
// SPDX-License-Identifier: Apache-2.0

#![no_std]
#![no_main]
#![feature(naked_functions, asm_sym)]

rust_syscall_tests::startup!();

use rust_syscall_tests::*;

fn main() -> Result<()> {
    let out = b"hi\n";
    let len = write(libc::STDERR_FILENO, out.as_ptr(), out.len())?;
    if len as usize == out.len() {
        Ok(())
    } else {
        Err(1)
    }
}
